client/query/query.go

```diff
@@ -127,6 +127,60 @@ func GetPaymentChannelInfo(id types.Destination, store store.Store, vm *payments
 	return PaymentChannelInfo{}, fmt.Errorf("could not find channel with id %v", id)
 }
 
+func GetAllLedgerChannels(store store.Store, consensusAppDefinition types.Address) ([]LedgerChannelInfo, error) {
+	toReturn := []LedgerChannelInfo{}
+
+	allConsensus, err := store.GetAllConsensusChannels()
+	if err != nil {
+		return []LedgerChannelInfo{}, err
+	}
+	for _, con := range allConsensus {
+		toReturn = append(toReturn, ConstructLedgerInfoFromConsensus(con))
+	}
+	allChannels := store.GetChannelsByAppDefinition(consensusAppDefinition)
+	for _, c := range allChannels {
+		toReturn = append(toReturn, ConstructLedgerInfoFromChannel(c))
+	}
+	return toReturn, nil
+}
+
+func GetPaymentChannelsByLedger(ledgerId types.Destination, store store.Store, vm *payments.VoucherManager) ([]PaymentChannelInfo, error) {
+	// If a ledger channel is actively funding payment channels it must be in the form of a consensus channel
```

Comment:
Are we 100% sure about this? I think recently we encountered some raciness around defunding a ledger channel (the point when the consensus channel is destroyed and replaced by a regular channel).